### PR TITLE
Directionnal adjacencies

### PR DESCRIPTION
**Pull request by stilnat**
# Summary

This PR adds directionnals connectables, such as sofas, booths and bench.

link to updated doc : https://ss3d.gitbook.io/design/environment/connectables/directional-adjacency-connections

#### PR checklist
- [x] The game builds properly without errors.

- [x] No unrelated changes are present.

- [x] No "trash" files are committed.

- [x] Relevant code is documented.

- [x] Update the related GitBook document, or create a new one if needed.

# Pictures/Videos)

https://github.com/RE-SS3D/SS3D/assets/14344825/5af0bd6c-5042-445a-8661-845a72c3daab

# Testing

#### Networking checklist

- [x] Works from host in host mode.

- [x] Works from server in server mode.

- [x] Works on server in client mode.

- [x] Works and is syncronized across different clients.

- [x] Is persistent.

# Changes

# Known issues

- There's still a few rare edge cases I did not precisely identify but saw while testing randomly. In some complex configurations, removing directionnals or loading back can cause bad rotation and/or shapes.  It's very rare though, and hard to identify in which configuration it occurs. I either need help to identify those, or we can merge and deal with them when they are better identified.

# Related issues/PRs

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
Are you reusing branches? I've been noticing excessive commits in your PRs lately. It's looks a lot cleaner and is easier to follow from a reviewer's perspective if you do not include not-related commits. Easiest way to do this that I'm aware of is by using a new branch for each PR.

**Comment by stilnat** [comment]
@cosmiccoincidence some of those branches, I've started not from develop but from other unmerged branches, because I needed what's on those other branches to work with. I thought it's not a big deal because all commits get squashed upon merge in develop, but you're right regarding reviewing. I usually don't review commit by commit myself that's why I did not think about it.
I'll try to find a solution to avoid that, but I'm a bit of a git noob and if you got any clues how to deal with that, that'd be appreciated.

**Comment by cosmiccoincidence** [comment]
> @cosmiccoincidence some of those branches, I've started not from develop but from other unmerged branches, because I needed what's on those other branches to work with. I thought it's not a big deal because all commits get squashed upon merge in develop, but you're right regarding reviewing. I usually don't review commit by commit myself that's why I did not think about it. I'll try to find a solution to avoid that, but I'm a bit of a git noob and if you got any clues how to deal with that, that'd be appreciated.

Normally I don't review commit by commit either, but sometimes I do take a quick look at commits for various reasons. For example, let's say I'm checking out the files changed page, and I see human.prefab had been edited, I can find the commit it was changed on and read the title of the commit or commit notes to help me figure out what changed if it wasn't obvious. If course I can look at the prefab in Unity, but sometimes looking at commits does help. 

If you need commits that haven't been merged then I can understand. But if you simply need commits that have been merged, make a new branch and make sure it's up to date with upstream.

File: Assets/Scripts/SS3D/Systems/Tile/Connections/AdjacencyTypes/DirectionnalShapeResolver.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace SS3D.Systems.Tile.Connections.AdjacencyTypes
+{
+    [Serializable]
+    public struct DirectionnalShapeResolver
```
Comment (cosmiccoincidence) [inline Assets/Scripts/SS3D/Systems/Tile/Connections/AdjacencyTypes/DirectionnalShapeResolver.cs:12]: "Directional" is spelt with a single n. Is the extra n in these script names intentional?

File: Assets/Content/WorldObjects/Furniture/Generic/Seats/DinnerBooth.prefab [hunk 2]
```diff
@@ -0,0 +1,142 @@
+%YAML 1.1
```
Comment (cosmiccoincidence) [inline Assets/Content/WorldObjects/Furniture/Generic/Seats/DinnerBooth.prefab:1]: - Add collision to this prefab
- Remove cast shadows from this prefab

**Review by cosmiccoincidence — CHANGES_REQUESTED** [review]
Tested it and it worked great from what I've experienced. I'm happy with the performance, any problematic edge cases can be identified in the future. Please add and link documentation for this prior to merge.

**Review by cosmiccoincidence — APPROVED** [review]
Amazing work!

**Pull request merged**